#168242071 Redirect users to deposits page upon successful payment

patrickf949 opened the pull request:
## Description ##
Users should be redirected to the deposits page upon making a successful payment where they can see their latest deposit.
User Pin should also be hidden and the input field should only be able to allow digits.

## Type of change ##
Please select the relevant option
- [x] Bug fix(a non-breaking change which fixes an issue)
- [ ] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [x] Integration test
- [ ] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## How can this PR be manually tested: ##
Make a payment using card & pin or a saved card and upon completion, the user should be redirected to the deposits page.


## Any background context you want to provide? ##
Users have been previously redirected to the home page, however, upon completion of the deposits feature it is only right that users who have successfully made deposits should be redirected to their deposits page with the latest deposit coming first in line.

## Related Pivotal Tracker stories ##
[#168242071](https://www.pivotaltracker.com/story/show/168242071)

wangonya reviewed (commented):
Hey @patrickf949, the code looks good but I'm having a little trouble testing this. Are there any particular credentials required for me to use while paying with Card and Pin? Maybe you could add a little bit of information on the `How to test` section.

Here's the error I'm getting `This is a test environment, only test cards can be used`

wangonya reviewed (approved):
Works as expected for the Card & Pin payment 🕺🏽

CryceTruly commented on package.json:
```diff
@@ -43,6 +43,7 @@
     "@nestjs/platform-express": "^6.0.0",
     "@nguniversal/express-engine": "^8.0.0",
     "@nguniversal/module-map-ngfactory-loader": "^8.0.0",
+    "@uiowa/digit-only": "^1.1.1",
```
Nice using this directive, saved some time

Tevinthuku reviewed (approved):
Nice work masking the pin inputs and making the accept only digits 👍

patrickf949 committed:
feat(redirect-to-deposits): redirect users to deposits page upon successful payment

- add route of deposits to the pin-validate service
- make sure pins are private
- add validation for input fields

[delivers #168242071]

kdaglas reviewed (approved):
LGTM

kdaglas reviewed (approved):
LGTM
